### PR TITLE
Refactor libauto_schedule.so Makefile

### PR DESCRIPTION
**Pull request by steven-johnson**
Move the rules for libauto_schedule.so (and friends) into autoscheduler.inc, so that the apps directories aren't dependent on apps/autoscheduler being built first. (Unbreaks 'make test_apps')

### CODE REVIEW

File: apps/support/autoscheduler.inc [hunk 1]
```diff
@@ -0,0 +1,68 @@
+AUTOSCHED_SRC=../autoscheduler
+
+WEIGHT_OBJECTS=\
+$(BIN)/weights_head1_conv1_weight.o \
```
Comment (abadams) [inline apps/support/autoscheduler.inc]: Doesn't the use of BIN here imply that a separate copy of the autoscheduler is built for each app? Should it be AUTOSCHED_BIN or something?

Comment (steven-johnson) [reply apps/support/autoscheduler.inc]: Yep, good point. Let me tweak.

Comment (steven-johnson) [reply apps/support/autoscheduler.inc]: PTAL

File: apps/random_pipeline/bench_comparison.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ SCHEDULES=1
 RANDOM_DROPOUT=100
 BEAM_SIZE=50
 
-make -C ../autoscheduler bin/libauto_schedule.so
+make -C ../autoscheduler ../autoscheduler/bin/libauto_schedule.so
```
Comment (abadams) [inline apps/random_pipeline/bench_comparison.sh]: This change looks fishy to me. make -C already sets the working directory to its argument.

Comment (steven-johnson) [reply apps/random_pipeline/bench_comparison.sh]: Will fix.

**Pull request merged**